aliyun/src/main/java/org/apache/iceberg/aliyun/AliyunProperties.java

```diff
@@ -25,21 +25,74 @@
 import org.apache.iceberg.util.PropertyUtil;
 
 public class AliyunProperties implements Serializable {
+  /**
+   * The domain name used to access OSS. OSS uses HTTP Restful APIs to provide services. Different regions are accessed
+   * by using different endpoints. For the same region, access over the internal network or over the Internet also uses
+   * different endpoints. For more information, see:
+   * https://www.alibabacloud.com/help/doc-detail/31837.htm
+   */
+  public static final String OSS_ENDPOINT = "oss.endpoint";
+
+  /**
+   * Aliyun uses an AccessKey pair, which includes an AccessKey ID and an AccessKey secret to implement symmetric
+   * encryption and verify the identity of a requester. The AccessKey ID is used to identify a user.
+   * <p>
+   * For more information about how to obtain an AccessKey pair, see:
+   * https://www.alibabacloud.com/help/doc-detail/53045.htm
+   */
+  public static final String ACCESS_KEY_ID = "access.key.id";
+
+  /**
+   * Aliyun uses an AccessKey pair, which includes an AccessKey ID and an AccessKey secret to implement symmetric
+   * encryption and verify the identity of a requester.  The AccessKey secret is used to encrypt and verify the
+   * signature string.
+   * <p>
+   * For more information about how to obtain an AccessKey pair, see:
+   * https://www.alibabacloud.com/help/doc-detail/53045.htm
+   */
+  public static final String ACCESS_KEY_SECRET = "access.key.secret";
+
+  /**
+   * Aliyun client factory for oss service, defaults to {@link DefaultAliyunClientFactory} implementation
+   */
+  public static final String CLIENT_FACTORY = "client.factory";
```

Comment:
This should be `client-factory-impl` to match conventions:
* Don't separate words with `.` -- `.` is used to separate hierarchy and `-` is used for multi-word concepts.
* `-impl` is used to specify an implementing class, like `fileio-impl` or `catalog-impl`

Reply:
Make sense ! 

Here we use the `client.factory` because we are following the naming style from `S3FileIO`.  I think it's better to name it to `client.factory-impl` in this PR.  Maybe we should propose another PR to address the S3 name also.